### PR TITLE
Improve STYLEGUIDE

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

This is a proposal to better organize the styleguide's sections and subsections, moving some stuff around, removing the use of bullet items when not needed, etc.

There is no significant content change proposed (yet).

### CODE REVIEW

File: STYLEGUIDE.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
 # Style Guide
 
-This is a general style guide that shall govern over the Flame Engine repository. The aim is to keep
-a common stable general environment. This includes high level guidance to help with simple decisions
+This is a general style guide that shall govern over all Flame repositories. The aim is to keep a
+common stable general environment. This includes high level guidance to help with simple decisions
```
Comment (wolfenrain) [inline STYLEGUIDE.md:4]: "common stable general" sounds strange to me :/ what are we trying to convey here?

File: STYLEGUIDE.md [hunk 2]
```diff
@@ -90,24 +92,39 @@ import 'game.dart';
...
+
+Example:
+
+````dart
+void something(int smaller, int bigger) {
+  assert(small < bigger, 'smaller is not smaller than bigger');
+  // ...
+}
+````
+## Comments
```
Comment (wolfenrain) [inline STYLEGUIDE.md:112]:
```suggestion

## Comments
```

**Review by wolfenrain — APPROVED** [review]

**Review by spydon — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

File: STYLEGUIDE.md (hunk 1 above)
Comment (luanpotter) [reply STYLEGUIDE.md:4]: i have no idea what I was thinking hahaha must have been a lapse.

Comment (luanpotter) [reply STYLEGUIDE.md]: AH that was already there haha. that explains why I don't know :P 

improved wording!

**Pull request merged**